client/my-sites/upgrades/domain-management/transfer/transfer-to-other-user/index.jsx

```diff
@@ -0,0 +1,218 @@
+
+		this.state = {
+			selectedUserId: '',
+			showConfirmationDialog: false,
+			disableDialogButtons: false
+		};
+
+		this.handleUserChange = this.handleUserChange.bind( this );
+		this.handleTransferDomain = this.handleTransferDomain.bind( this );
+		this.handleDialogClose = this.handleDialogClose.bind( this );
+		this.handleConfirmTransferDomain = this.handleConfirmTransferDomain.bind( this );
+	}
+
+	componentWillUpdate( nextProps, nextState ) {
+		if ( nextState && ! nextState.selectedUserId ) {
+			const defaultUser = head( this.filterAvailableUsers( nextProps.users ) );
+			if ( defaultUser ) {
+				this.setState( { selectedUserId: defaultUser.ID } );
+			}
+		}
+	}
+
+	handleUserChange( event ) {
+		event.preventDefault();
+
+		this.setState( { selectedUserId: event.target.value } );
+	}
+
+	handleTransferDomain() {
+		this.setState( { showConfirmationDialog: true } );
+	}
+
+	handleConfirmTransferDomain( closeDialog ) {
+		const { selectedDomainName } = this.props,
+			selectedUserDisplay = this.getSelectedUserDisplayName(),
+			successMessage = this.props.translate(
+				'%(selectedDomainName)s has been transferred to %(selectedUserDisplay)s',
+					{ args: { selectedDomainName, selectedUserDisplay } } ),
+			defaultErrorMessage = this.props.translate(
+				'Failed to transfer %(selectedDomainName)s, please try again or contact support.', {
+					args: { selectedDomainName } } );
+
+		this.setState( { disableDialogButtons: true } );
+		wpcom.transferToUser( this.props.selectedSite.ID, this.props.selectedDomainName, this.state.selectedUserId )
+			.then( () => {
+				this.setState( { disableDialogButtons: false } );
+				this.props.successNotice( successMessage, { duration: 4000, isPersistent: true } );
+				page( paths.domainManagementEdit( this.props.selectedSite.slug, this.props.selectedDomainName ) );
+				closeDialog();
+			}, err => {
+				this.setState( { disableDialogButtons: false } );
+				this.props.errorNotice( err.message || defaultErrorMessage );
+				closeDialog();
+			} );
+	}
+
+	handleDialogClose() {
+		if ( ! this.state.disableDialogButtons ) {
+			this.setState( { showConfirmationDialog: false } );
+		}
+	}
+
+	getSelectedUserDisplayName() {
+		const selectedUser = find( this.props.users, user => user.ID === Number( this.state.selectedUserId ) );
+		if ( ! selectedUser ) {
+			return '';
+		}
+		return this.getUserDisplayName( selectedUser );
+	}
+
+	getUserDisplayName( { first_name, last_name, nice_name } ) {
+		return ( first_name && last_name ) ? `${ first_name } ${ last_name } (${ nice_name })` : nice_name;
+	}
+
+	render() {
+		if ( ! this.isDataReady() ) {
+			return <DomainMainPlaceholder goBack={ this.goToEdit } />;
+		}
+
+		const { selectedSite, selectedDomainName } = this.props,
+			{ slug } = selectedSite;
+
+		return (
+			<Main className="transfer-to-other-user">
+				<Header
+					selectedDomainName={ selectedDomainName }
+					backHref={ paths.domainManagementEdit( slug, selectedDomainName ) }>
```

Comment:
This should be `paths.domainManagementTransfer` - otherwise, we "jump" back two screens.